### PR TITLE
Add release job

### PR DESCRIPTION
**Pull request by owanturist**
Add a step to CI for creating a release PR or publishing the released changes

### CODE REVIEW

File: .github/workflows/build-extension-charts.yml [hunk 1]
```diff
@@ -11,13 +11,26 @@ defaults:
...
+    needs: prepare
     permissions:
       actions: write
       contents: write
       deployments: write
       pages: write
     with:
       target_branch: gh-pages
-      tagged_release: ${{ github.ref_name }}
+      tagged_release: ${{ needs.prepare.outputs.tagged_release }}
```
Comment (owanturist) [inline .github/workflows/build-extension-charts.yml:36]: the `tagged_release` expects `PACKAGE_NAME-VERSION` format but changesets sets `PACKAGE_NAME@VERSION` when runs the release command, so it should be pre-formatted

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -20,16 +24,23 @@ jobs:
...
           node-version-file: .nvmrc
-          cache: "yarn"
+          cache: yarn
 
       - name: Install dependencies
         run: yarn install --frozen-lockfile --prefer-offline
 
+      - name: Check for uncommitted changes
+        run: |
+          git diff --name-only --exit-code --relative || (echo "\nUncommitted changes found up there 👆" && exit 1)
```
Comment (owanturist) [inline .github/workflows/checks.yml:42]: it verifies that there are no uncommited yarn.lock changes

File: .github/workflows/build-extension-charts.yml [hunk 3]
```diff
@@ -11,13 +11,26 @@ defaults:
...
+      - name: Format release details
+        id: release-details
+        # Replace @ with -
+        run: |
+          TAG="${${{ github.ref_name }}//@/-}" 
+          echo "tagged_release=$TAG" >> $GITHUB_OUTPUT
+
   build-extension-charts:
-    uses: rancher/dashboard/.github/workflows/build-extension-charts.yml@master
+    uses: rancher/dashboard/.github/workflows/build-extension-charts.yml@release-2.8
```
Comment (owanturist) [inline .github/workflows/build-extension-charts.yml:27]: the workflow must be compatible with the supported extension version

File: .github/workflows/checks.yml (hunk 2 above)
Comment (craffit) [reply .github/workflows/checks.yml:41]: nice

**Comment by owanturist** [comment]
Closing this PR in favor of #41

**Pull request closed without merging**